### PR TITLE
Update to Angular 16

### PR DESCRIPTION
**Pull request by samiv22**
Hi,

I've found this package to be most helpful in my current project but as I started to update my Angular-version in my project I noticed that this does not support >Angular 15.

I've tried to make this now work with Angular 16, but this is my first ever contribution so please provide feedback if I missed anything!

- Updated Changelog to include version 9 and 10 changes
- Added compatibility section to readme.md
- Fixed couple of errors my IDE (PHPStorm) gave
- Updated Angular version from 15->16
- Checked that all tests pass. 1 test is skipped.

Will be creating also Angular version 17 update if this goes well :)

Br, Sami

### CODE REVIEW

File: projects/test-app/src/app/app.component.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Component } from '@angular/core';
 import { UntypedFormBuilder } from '@angular/forms';
-import { Validators } from 'angular-reactive-validation';
+import { Validators } from '../../../angular-reactive-validation/src/lib/validators';
```
Comment (davidwalschots) [inline projects/test-app/src/app/app.component.ts]: **thought**: It's not necessary to change this path, as the original code will work after executing `ng build angular-reactive-validation`. That is a bit unclear, so I'll try to make it clearer in a separate PR.

File: projects/test-app/src/app/app.component.ts [hunk 2]
```diff
@@ -15,15 +15,15 @@ export class AppComponent {
     name: this.fb.group({
       firstName: ['', [Validators.required('A first name is required'),
         Validators.minLength(5),
-        Validators.maxLength(10, (maxLength => `Maximum length is ${maxLength}`))]],
-      middleName: ['', [Validators.maxLength(50, (maxLength => `Maximum length is ${maxLength}`))]],
+        Validators.maxLength(10, ((maxLength: any) => `Maximum length is ${maxLength}`))]],
```
Comment (davidwalschots) [inline projects/test-app/src/app/app.component.ts]: **thought**: As the type of `maxLength` is already known, it's not necessary to specify the `any` type here. 

PHPStorm seems wrong here. This is what I see in vscode:

image

File: projects/test-app/src/app/app.module.ts [hunk 3]
```diff
@@ -1,12 +1,10 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
 import { ReactiveFormsModule } from '@angular/forms';
-import { ReactiveValidationModule } from 'angular-reactive-validation';
-
+import { ReactiveValidationModule } from '../../../angular-reactive-validation/src/lib/reactive-validation.module';
```
Comment (davidwalschots) [inline projects/test-app/src/app/app.module.ts]: **thought**: It's not necessary to change this path, as the original code will work after executing `ng build angular-reactive-validation`. That is a bit unclear, so I'll try to make it clearer in a separate PR.

File: projects/angular-reactive-validation/package.json [hunk 4]
```diff
@@ -19,10 +19,10 @@
...
   "private": false,
   "peerDependencies": {
-    "@angular/common": "^15.0.0",
-    "@angular/core": "^15.0.0",
-    "@angular/forms": "^15.0.0",
-    "rxjs": "^6.5.3 || ^7.4.0"
+    "@angular/common": "^16.0.0",
+    "@angular/core": "^16.0.0",
+    "@angular/forms": "^16.0.0",
+    "rxjs": "^6.5.3 || ^7.8.0"
```
Comment (davidwalschots) [inline projects/angular-reactive-validation/package.json]: **note**: According to [this table](https://angular.io/guide/versions#actively-supported-versions) Angular 16 supports rxjs `^6.5.3 || ^7.4.0`.
```suggestion
    "rxjs": "^6.5.3 || ^7.4.0"
```

File: README.md [hunk 5]
```diff
@@ -21,8 +22,15 @@ To install this library, run:
 npm install angular-reactive-validation --save
 ```
 
-## Basic usage
+## Compatibility
```
Comment (davidwalschots) [inline README.md:25]: This table is a nice addition!

**Review by davidwalschots — COMMENTED** [review]
Thank you for your contribution! A few comments, but nothing major.

File: projects/angular-reactive-validation/package.json (hunk 4 above)
Comment (samiv22) [reply projects/angular-reactive-validation/package.json]: Thanks!

File: README.md (hunk 5 above)
Comment (samiv22) [reply README.md:25]: Thank you, I always find these helpful

File: projects/test-app/src/app/app.component.ts (hunk 1 above)
Comment (samiv22) [reply projects/test-app/src/app/app.component.ts]: I've reverted this line.

File: projects/test-app/src/app/app.module.ts (hunk 3 above)
Comment (samiv22) [reply projects/test-app/src/app/app.module.ts]: I've reverted this change.

File: projects/test-app/src/app/app.component.ts (hunk 2 above)
Comment (samiv22) [reply projects/test-app/src/app/app.component.ts]: Reverted 👍

**Comment by davidwalschots** [comment]
I'll release an npm package of this version somewhere tomorrow. Should you be done with the Angular 17 PR by that time I'll also release that one. Thanks again for your contribution! 🚀

**Comment by davidwalschots** [comment]
I am trying to publish this package, but something seems amiss with npmjs.com's authentication. I'm being redirected to a 404 in the authentication flow. Hopefully they'll fix this soon? :D

**Pull request merged**